### ISSUE TITLE
Segmentation fault when running effil.thread

### ISSUE THREAD

**Issue by sam01101** [effil/effil#153]
Problem: Executing the following code will cause a "Segmentation fault" error.

Expected: Code run successfully
Reuslt: A "Segmentation fault" displayed.

How to trigger the error:

- Run the following code

(I've also tested without looping, there still have a chance will trigger the error.)

Lua version: 
- openresty luajit 2.1.0-beta3
- luajit 2.0
- Lua 5.1
(All are tested and reuslting the error)

Platform:
Raspberry pi 4
`Linux raspberrypi 5.4.51-v8+ #1327 SMP PREEMPT aarch64 GNU/Linux`

Code:
```lua
local effil = require"effil"
local a = function() end
local b = function() a() end
while true do
    effil.thread(function() a() b() end)()
end

**Comment by mihacooper** [comment]
Great thanks for this issue!
It seems to be bug that I'm looking for a long time)
I have made a fix and I hope we'll release it soon:
https://github.com/effil/effil/issues/153

**Comment by buzz3rs** [comment]
I did encounter a similar error before but it didn’t give me a “segmentation fault” and avoiding this i defined the functions in the global effil.G, worked fine for me

**Comment by buzz3rs** [comment]
I did encounter a similar error before but it didn’t give me a “segmentation fault” and avoiding this i defined the functions in the global effil.G, worked fine for me

**Comment by buzz3rs** [comment]
I did encounter a similar error before but it didn’t give me a “segmentation fault” and avoiding this i defined the functions in the global effil.G, worked fine for me

**Comment by mihacooper** [comment]
Fixes in release v1.2-0
